packages/cli/src/build/__tests__/other.ts

```diff
@@ -241,6 +268,15 @@ test("typescript", async () => {
     export default thing;
     export { obj };
 
+    ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ dist/typescript.esm.js ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
+    'use strict';
+
+    if (process.env.NODE_ENV === "production") {
+      module.exports = require("./typescript.esm.prod.js");
+    } else {
+      module.exports = require("./typescript.esm.dev.js");
+    }
+
```

Comment:
We definitely don't want to be emitting CJS for ESM

Reply:
This is a challenge because there isn't an equivalent (that I can think of) for doing this in pure ESM. What bundle should we pick if bundler doesn't support "production" or "development" conditionals? Always prod?

Reply:
For this PR, I'd say let's not have dev and prod bundles for ESM. The only difference between the dev and prod bundles is that `process.env.NODE_ENV` is replaced with `"production"` and then Rollup does some dead code elimination based on that (note that it's _not_ replaced with `"development"` in the dev bundle). The background behind the dev and prod bundles is that accessing `process.env.NODE_ENV` in Node is, as you would kind of expect, not super fast so it's helpful to replace it so that we just check it once per entrypoint in Node rather on potentially on some very hot paths, and since currently our ESM is targeting bundlers where `process.env.NODE_ENV` will be replaced statically, providing dev and prod bundles doesn't really benefit much.

Reply:
I'm thinking that we can just remove the "development" condition and then default "esm" to always use the "dev" build (since it's mostly untouched). This way we're always sending ESM when "module" condition is chosen and users can opt into "production" builds by setting their "NODE_ENV" or specifying their preference on the "production" conditional (some bundlers do this automatically for you). The exports field would look something like this: 

```{
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "worker": {
        "production": {
          "module": "./dist/emotion-react.worker.esm.prod.js",
          "default": "./dist/emotion-react.worker.cjs.prod.js"
        },
        "module": "./dist/emotion-react.worker.esm.dev.js",
        "default": "./dist/emotion-react.worker.cjs.js"
      },
      "browser": {
        "production": {
          "module": "./dist/emotion-react.browser.esm.prod.js",
          "default": "./dist/emotion-react.browser.cjs.prod.js"
        },
        "module": "./dist/emotion-react.browser.esm.dev.js",
        "default": "./dist/emotion-react.browser.cjs.js"
      },
      "production": {
        "module": "./dist/emotion-react.esm.prod.js",
        "default": "./dist/emotion-react.cjs.prod.js"
      },
      "module": "./dist/emotion-react.esm.dev.js",
      "default": "./dist/emotion-react.cjs.js"
    },
  }
}

Note that the `default` condition remains the same so it will still include the runtime "NODE_ENV" check and dynamically require the correct bundle.

Reply:
This change has been implemented. This only potential issue with this that I can think of is that people might "accidentally" use the dev version when using a bundler that doesn't replace "NODE_ENV" or set the "production" conditional preference (eg. esbuild doesn't not do this automatically).

Reply:
@mitchellhamilton just rechecking here - this already seems to be addressed and looks OK to me, could you confirm that you have no further comments on this one?